src/votekit/elections/election_types.py

```diff
@@ -1117,3 +1120,317 @@ def __init__(
             seats=seats,
             tiebreak=tiebreak,
         )
+
+
+class RandomDictator(Election):
+    """
+    Choose a winner randomly from the distribution of first place votes. For multi-winner elections
+    repeat this process for every winner, removing that candidate from every voter's ballot
+    once they have been elected.
+
+    Args:
+      profile (PreferenceProfile): PreferenceProfile to run election on.
+      seats (int): Number of seats to elect.
+
+    Attributes:
+      _profile (PreferenceProfile): PreferenceProfile to run election on.
+      seats (int): Number of seats to be elected.
+
+    """
+
+    def __init__(self, profile: PreferenceProfile, seats: int):
+        super().__init__(profile, ballot_ties=False)
+        self.seats = seats
+
+    def next_round(self) -> bool:
+        """
+        Determines if another round is needed.
+
+        Returns:
+            (bool) True if number of seats has not been met, False otherwise.
+        """
+        cands_elected = [len(s) for s in self.state.winners()]
+        return sum(cands_elected) < self.seats
+
+    def run_step(self):
+        if self.next_round():
+            remaining = self.state.profile.get_candidates()
+
+            ballots = self.state.profile.ballots
+            weights = [b.weight for b in ballots]
+            random_ballot = random.choices(
+                self.state.profile.ballots, weights=weights, k=1
+            )[0]
+
+            # randomly choose a winner from the first place rankings
+            winning_candidate = random.choice(list(random_ballot.ranking[0]))
+
+            # some formatting to make it compatible with ElectionState, which
+            # requires a list of sets of strings
+            elected = [{winning_candidate}]
+
+            # remove the winner from the ballots
+            new_ballots = remove_cand(winning_candidate, self.state.profile.ballots)
+            new_profile = PreferenceProfile(ballots=new_ballots)
+
+            # determine who remains
+            remaining = [{c} for c in remaining if c != winning_candidate]
+
+            # update for the next round
+            self.state = ElectionState(
+                curr_round=self.state.curr_round + 1,
+                elected=elected,
+                eliminated_cands=[],
+                remaining=remaining,
+                profile=new_profile,
+                previous=self.state,
+            )
+
+            # if this is the last round, move remaining to eliminated
```

Comment:
Let's keep all the remaining candidates as remaining, since they were not technically eliminated in your election process.